frontend/src/concepts/k8s/K8sNameField.tsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import {
+  TextInput,
+  FormGroup,
+  Tooltip,
+  Stack,
+  StackItem,
+  HelperText,
+  HelperTextItem,
+  ValidatedOptions,
+  HelperTextItemProps,
+} from '@patternfly/react-core';
+import { HelpIcon } from '@patternfly/react-icons';
+import { NameDescType, K8sNameValidationState } from '~/pages/projects/types';
+import { isK8sNameValid } from '~/pages/projects/utils';
+
+type K8sNameFieldProps = {
+  data: NameDescType;
+  setData: (data: NameDescType) => void;
+  nameFieldId: string;
+  disableK8sName?: boolean;
+  validateK8sName: K8sNameValidationState;
+  setValidateK8sName: React.Dispatch<React.SetStateAction<K8sNameValidationState>>;
+};
+
+const K8sNameField: React.FC<K8sNameFieldProps> = ({
+  nameFieldId,
+  disableK8sName,
+  data,
+  setData,
+  validateK8sName,
+  setValidateK8sName,
+}) => {
+  const { ruleLength, ruleCharacterTypes } = validateK8sName;
+
+  const handlek8sNameChange = (k8sName: string) => {
+    setData({
+      ...data,
+      k8sName: { value: k8sName, isTruncated: false, isUserInputk8sName: true },
+    });
+
+    if (k8sName.length >= 30) {
+      setValidateK8sName({
+        ruleCharacterTypes: !isK8sNameValid(k8sName) ? 'error' : 'success',
+        ruleLength: 'warning',
+      });
+      setData({
+        ...data,
+        k8sName: { value: k8sName.slice(0, 30), isTruncated: true, isUserInputk8sName: true },
+      });
+    } else if (data.k8sName?.value.length === 0) {
+      setValidateK8sName({
+        ruleCharacterTypes: 'indeterminate',
+        ruleLength: 'indeterminate',
+      });
+    } else {
+      setValidateK8sName({
+        ruleCharacterTypes: !isK8sNameValid(k8sName) ? 'error' : 'success',
+        ruleLength: 'success',
+      });
+    }
+  };
+
+  return (
+    <FormGroup
+      label="Resource name"
+      labelIcon={
+        <Tooltip
+          removeFindDomNode
+          position="right"
+          content={
+            <Stack hasGutter>
+              <StackItem>
+                Resource names are what your resources are labeled in OpenShift.
+              </StackItem>
+              <StackItem>Resource names are not editable after creation.</StackItem>
```

Comment:
I think this should be combined in one sentence. @xianli123 can you confirm? For reference: https://github.com/opendatahub-io/odh-dashboard/issues/1627#issuecomment-1736257017 and https://github.com/opendatahub-io/odh-dashboard/issues/1627#issuecomment-1736593536

@xianli123 FYI the tooltip looks like this right now: 
<img width="323" alt="Screenshot 2023-10-27 at 4 15 01 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/22885912/af3d8aa5-4548-4e2a-b58f-5e15d7b4c7d9">

Reply:
@manaswinidas Yes, these 2 sentences should be combined.

Reply:
@uidoyen can you change the text to this? "The resource name is used to identify your resource in OpenShift, and is not editable after creation."